x-pack/plugins/maps/public/shared/layers/styles/components/vector/legend/vector_icon.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { dynamicColorShape, staticColorShape } from '../style_option_shapes';
+import { FillableCircle, FillableVector } from '../../../../../icons/additional_layer_icons';
```

Comment:
while we're in here, consider renaming `FillableVector` to `FillableRectangle`, because that's what it is https://github.com/thomasneirynck/kibana/blob/52fed8a52d0124ba454652d540d10ad2592605ad/x-pack/plugins/maps/public/shared/icons/additional_layer_icons.js#L47